### DISCUSSION TITLE
Opencart integration

### DISCUSSION THREAD

**Discussion by BaiMangal** [btcpayserver/btcpayserver#870] [Plugins & integrations]
**Describe the solution you'd like**
would be nice to have a plugin for https://www.opencart.com the same way coingate and bitpay have

**Describe alternatives you've considered**
to use coingate or bitpay

**Comment by atoshk** [comment]
why not have opencart plugins? opencart already have many users. more than prestashop or wordspress. we are waiting opencart plugin for btcpay server as soon as possible

**Comment by demianrihs** [comment]
I am also interested in this

**Comment by dstrukt** [comment]
cc @woutersamaey

**Reply by woutersamaey** [reply]
If enough people are interested, we can definitely consider this.

**Reply by pavlenex** [reply]
Agreed, why not.

**Reply by woutersamaey** [reply]
@ndeet have you looked at OpenCart before?

**Reply by ndeet** [reply]
@woutersamaey @pavlenex no, I heard of it but never installed or used used it. According to some statistics they have about 580k installations, which seems to be much less than WooCommerce or Magento but I'm open to it if there is enough demand. 

@pavlenex maybe we do a poll via BTCPay twitter account? 

@woutersamaey The installation instructions are interesting, they talk about a composer version but there is no composer.json to be found, so you basically just copy contents of the "upload" folder to your webhost. In the installation docs they link to a module skeleton generator that leads to a 404 http://opencart.hostjars.com/creating-opencart-modules 😬 

Anyway, seems to be maintained and if there is enough demand we can look into it if there is enough demand. Question is what number of interested people makes sense? Above we have 4, so maybe a BTCPay Server twitter poll makes sense? Maybe even give choice for 2 other PHP popular shopping cart systems (we have no integrations with yet) to see relative demand too? (or see if there is a even more requested PHP store system out there)

**Reply by pavlenex** [reply]
@ndeet Good idea, let's make a poll, which other two php store systems exist?

**Reply by woutersamaey** [reply]
- Magento 1 / OpenMage
- Shopware
- Drupal Commerce

Coming up with more names is getting hard though... we have Prestashop right? Not sure if it's upgraded to Greenfield yet. Magento 1 we have, but old module and also no Greenfield.

**Reply by pavlenex** [reply]
We do have Magento 1 and Drupal, do you mean we should re-write them to use greenfield?
- https://docs.btcpayserver.org/Magento/#magento-1
- https://docs.btcpayserver.org/Drupal/
- Presta is using greenfield.

**Reply by pavlenex** [reply]
Posted https://twitter.com/BtcpayServer/status/1499337421808357383?s=20&t=xykAM-mjxzZ1LKu8-mb7gA

**Reply by Alentia** [reply]
I need integration of BTCPay and OpenCart.

**Reply by Alentia** [reply]
There are only 100K live Magento sites vs 450K Opencart sites
![image](https://user-images.githubusercontent.com/99363901/162597409-e718b870-09c9-42ed-8dff-6a1d44e1f9e4.png)
![image](https://user-images.githubusercontent.com/99363901/162597425-deaf2f97-9537-486d-a00b-d7e7628a90ef.png)

**Comment by zaqueoae** [comment]
Hi,
I am also interested in implementing btcpay for all my clients using opencart. Would be wonderful.

**Reply by ndeet** [reply]
Hi, I just finished the work on the Joomla integration and now started with OpenCart, will take a couple of weeks but it is in the works! 🚀

**Comment by ndeet** [comment]
Hi, the OpenCart 3.x extension is now available for download. Would be great if some of you running OpenCart could test it out.

I'm still waiting for approval to get listed on their marketplace cloud which would make the extension installable from the admin dashboard but by downloading and uploading to your store it should also work. 

Marketplace entry:
https://www.opencart.com/index.php?route=marketplace/extension/info&extension_id=44269&filter_category_id=3

Github repository, please report all bugs there: 
https://github.com/btcpayserver/opencart

Documentation preview:
https://github.com/btcpayserver/btcpayserver-doc/blob/555a5d3f886bff0ec008a74d763b9f96b6473ce6/docs/OpenCart.md

I saw that in the meantime OpenCart 4.x was released and asked the maintainer if we can make it compatible with the same codebase which he said it is better to have separate codebase. Unfortunately their documentation is very bad even for OC 3. So I will need to see on their core extensions how things are supposed to be structured etc. From what I saw with other extensions it should be possible to use what we have and make it work with OC 4.x. Will have a separate branch to do two release channels.

**Reply by ndeet** [reply]
Hi again, after lots of try and error and reverse engineering as there is even less OpenCart 4 documentation here is the OC4 release:

https://github.com/btcpayserver/opencart/releases/tag/v4.0.0 or 
https://www.opencart.com/index.php?route=marketplace/download&extension_id=44269

Docs preview (same as for OC3) https://github.com/btcpayserver/btcpayserver-doc/blob/47889add30dd2ff188db5fc4175c7489dee8965a/docs/OpenCart.md

**Reply by ndeet** [reply]
Documentation now merged and available https://docs.btcpayserver.org/OpenCart/ 🎉
